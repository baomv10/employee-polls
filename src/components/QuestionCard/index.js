import { format } from "date-fns";
import { useNavigate } from "react-router-dom";

const QuestionCard = ({ question }) => {
  const navigate = useNavigate();

  return (
    <div
      className="col-3 border rounded m-3 p-3 text-center"
      key={question?.id}
    >
      <h5>{question?.author}</h5>
      <p>{format(new Date(question?.timestamp || 0), "h:mm a | MM/dd/yyyy")}</p>
      <div>
        <div className="d-grid">
          <button
            className="btn btn-outline-success"
            onClick={() => navigate(`questions/${question?.id}`)}
          >
            Show
          </button>
        </div>
      </div>
    </div>
  );
};

export default QuestionCard;
