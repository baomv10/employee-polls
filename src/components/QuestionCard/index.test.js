import * as React from "react";
import { render } from "@testing-library/react";
import QuestionCard from ".";
import { BrowserRouter } from "react-router-dom";

describe("QuestionCard", () => {
  it("matches the snapshot when question is passed", () => {
    var component = render(
      <BrowserRouter>
        <QuestionCard
          question={{
            id: "8xf0y6ziyjabvozdd253nd",
            author: "sarahedo",
            timestamp: 1467166872634,
            optionOne: {
              votes: ["sarahedo"],
              text: "Build our new application with Javascript",
            },
            optionTwo: {
              votes: [],
              text: "Build our new application with Typescript",
            },
          }}
        />
      </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
  });

  it("matches the snapshot when no question is passed", () => {
    var component = render(
      <BrowserRouter>
        <QuestionCard />
      </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
