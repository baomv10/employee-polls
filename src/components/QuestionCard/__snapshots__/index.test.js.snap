// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCard matches the snapshot when no question is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-3 border rounded m-3 p-3 text-center"
      >
        <h5 />
        <p>
          8:00 AM | 01/01/1970
        </p>
        <div>
          <div
            class="d-grid"
          >
            <button
              class="btn btn-outline-success"
            >
              Show
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-3 border rounded m-3 p-3 text-center"
    >
      <h5 />
      <p>
        8:00 AM | 01/01/1970
      </p>
      <div>
        <div
          class="d-grid"
        >
          <button
            class="btn btn-outline-success"
          >
            Show
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`QuestionCard matches the snapshot when question is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-3 border rounded m-3 p-3 text-center"
      >
        <h5>
          sarahedo
        </h5>
        <p>
          9:21 AM | 06/29/2016
        </p>
        <div>
          <div
            class="d-grid"
          >
            <button
              class="btn btn-outline-success"
            >
              Show
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-3 border rounded m-3 p-3 text-center"
    >
      <h5>
        sarahedo
      </h5>
      <p>
        9:21 AM | 06/29/2016
      </p>
      <div>
        <div
          class="d-grid"
        >
          <button
            class="btn btn-outline-success"
          >
            Show
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
